Source/Provers/SMTLib/SMTLibInteractiveTheoremProver.cs

```diff
@@ -29,6 +29,10 @@ public class SMTLibInteractiveTheoremProver : SMTLibProcessTheoremProver
 
     public override Task GoBackToIdle()
     {
+      if (Process == null) {
```

Comment:
When does this happen?

Reply:
I have no idea but it did happen on my machine and caused a crash as I was debugging the Dafny Language Server and heavily loaded it with a lot of requests.
The only reasons why `Process` would be null would be
- If it `GoBackToIdle()` was called before a single call to `SetupProcess()`
- If the method `Close()` was called before `GoBackToIdle()`

It would be worth investigating why, but for now, I found it should be sound to do this change, because if the process is null, it means it was either not started or it was closed, so it can always be restarted and the prover can be reused.

Reply:
Just added a test that fails by replicating the behavior "close()" before "GoBackToIdle()". It's not the panacea, as in the first place we should never have such a call and there might be some more lurking bugs. But it's better than nothing.